daemon/algod/api/spec/v2/model.go

```diff
@@ -22,25 +22,37 @@ import (
 	"github.com/algorand/go-algorand/ledger/ledgercore"
 )
 
-// AccountResourceModel used to encode AccountResource
-type AccountResourceModel struct {
+// AccountAssetModel is returned by AccountAssetInformation when msgpack format is specified
+type AccountAssetModel struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	AssetParams  basics.AssetParams  `codec:"asset-params"`
+	AssetHolding basics.AssetHolding `codec:"asset-holding"`
+}
+
+// AccountApplicationModel is returned by AccountApplicationInformation when msgpack format is specified
+type AccountApplicationModel struct {
 	_struct struct{} `codec:",omitempty,omitemptyarray"`
 
-	AssetParams   basics.AssetParams   `codec:"asset-params"`
-	AssetHolding  basics.AssetHolding  `codec:"asset-holding"`
 	AppLocalState basics.AppLocalState `codec:"app-local-state"`
 	AppParams     basics.AppParams     `codec:"app-params"`
```

Comment:
Unrelated to this change, it would be more informative if these types were pointers. The reason being the zero values for `basics.AppLocalState` and `basics.AssetHolding` still represent real values, so you can't determine if the field was present or not based on this structure.

Reply:
Yes, I agree. we need to make sure we can clearly differentiate between the default basics.AssetHolding to the absence of that field. ( i.e. which makes the difference of whether the asset is being opted in or not )

Reply:
OK, split it up, and updated the calling code in goal to check nil (similar to how it did before https://github.com/algorand/go-algorand/pull/3605/files#diff-b9b58335b174b56cfb0585bb4281c30deea9b3ffb0a129fba96f641933a5ab88)